main.c

```diff
@@ -197,14 +219,31 @@ asm_return_t lz_main(void)
 	*tb_dev_map = _u(dev_table);
 
 	code32_start = zero_page + BP_CODE32_START;
-	slaunch_header_offset = zero_page + BP_MLE_HEADER;
+	print("\ncode32_start ");
+	print_p(_p(*code32_start));
+
+	if (*((u16*)(zero_page + BP_VERSION)) >= 0x020f) {
+		u32 *kernel_info_offset;
+		kernel_info_offset = zero_page + BP_KERNEL_INFO;
+		print("\nkernel_info ");
+		print_p(_p(*kernel_info_offset));
+		slaunch_header_offset = _p(*code32_start + *kernel_info_offset + KI_MLE_HEADER);
+		print("\nslaunch_hdr_offset ");
+		print_p(_p(*slaunch_header_offset));
+	} else {
+		slaunch_header_offset = zero_page + BP_KERNEL_INFO;
```

Comment:
So there never really will be a kernel that specifies the MLE offset like this in the real world. Maybe this should just be an error condition (print and exit).

Reply:
We needed it for stuff we produced internally that was still floating around, but as we have a tag now, I think I can change it.

Reply:
Yea that is kind of what I figured. Thanks.

Reply:
Done